index.js

```diff
@@ -233,7 +237,29 @@ module.exports = function (options) {
 
   router.get('/', function (req, res) {
     // If we've gotten this far then token gives proper access and that's all that this route care about
-    return res.sendStatus(200);
+    var isSyndicationRequest = (req.query.q === 'syndicate-to');
```

Comment:
Continuing on the thought of using a callback that I posted in the previous comment – I would then replace this here with a call to that callback – so something like:

```javascript
if (req.query.q) {
  if (!options.queryHandler) { return badRequest(res, 'Queries not supported'); }
  options.queryHandler(req.query.q.trim().toLowerCase(), req).then(function (result) {
    // Handle the result
  })
}
```